parity-clib/examples/cpp/main.cpp

```diff
@@ -14,172 +14,134 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
 
+#ifdef __clang__
+#pragma clang diagnostic error "-Wvexing-parse"
```

Comment:
Do we require this? Are we dedicated to using `clang` ?

Reply:
We do not require this.  I use `clang` myself, but this code should work with any conforming C++17 compiler.  This code is harmless, but I can go ahead and remove it if you want.

Reply:
Unknown pragmas are going to be ignored by other compilers. So it doesn't hurt. I just wonder if we could encode this behaviour into CMake maybe?

Reply:
Not really.  `__clang__` will only be defined by clang, so this will certainly be ignored by other compilers.